packages/grafana-ui/src/components/DataTable/types.ts

```diff
@@ -0,0 +1,34 @@
+import { ReactNode } from 'react';
+import { CellProps, DefaultSortTypes, IdType, SortByFn } from 'react-table';
+
+export interface Column<TableData extends object> {
+  /**
+   * ID of the column.
+   * Set this to the matching object key of your data or `undefined` if the column doesn't have any associated data with it.
+   * This must be unique among all other columns.
+   */
+  id?: IdType<TableData>;
```

Comment:
technically it is used to match the column with the relevant piece of data. setting it to undefined will cause the underlying data to not match with any column, but i agree is confusing. may it's better to just make it required? if the key does not exist in the underlying data there won't be any issue, the only reason i made it option was to avoid users to have to define an id when they didn't need it, but again, maybe making it required would simplify reasoning about it. wdyt?

Reply:
Yeah I don't think it's that much worse to force every column to have an id and it might save us from other issues in the future so making it required sounds like a good idea!

Reply:
changed in 2a8aceaed5